airflow/contrib/executors/kubernetes_executor.py

```diff
@@ -694,8 +698,8 @@ def start(self):
         # https://github.com/kubernetes-client/python/blob/master/kubernetes/docs
         # /CoreV1Api.md#list_namespaced_pod
         KubeResourceVersion.reset_resource_version()
-        self.task_queue = Queue()
-        self.result_queue = Queue()
+        self.task_queue = self._manager.Queue()
```

Comment:
And here maybe we could use a multiprocessing.JoinableQueue instead.

Reply:
Agree. I think in this case we have no empty() problem.

Reply:
I also think we should get rid of the qsize() as it won't work on MacOS

Reply:
Ok. I took a deeper look and I discovered something else. Unlike the original multiprocessing.Queue(), the queue() returned by manager's Queue() method is actually a joinable queue already (both Python 2 and 3). And if I tried to use a direct (non-managed) Joinable Queue with get_nowait() I hit the same problem as the original LocalExecutor.test().\

At the end I am changing it to manager.Queue() and add task_done() to handle join separately.

BTW. I think task_done() was missing in the original implementation anyway as already .join() was used which could have caused unclean shutdown/hangs in the kubernetes executor. I kind of wonder how it worked before at all.